spec/services/courses/generate_course_title_service_spec.rb

```diff
@@ -0,0 +1,130 @@
+
+      it "Appends SEND information to the title" do
+        expect(generated_title).to end_with("with Special educational needs and disability")
+      end
+    end
+  end
+
+  context "With no subjects" do
+    it "Returns an empty string" do
+      expect(generated_title).to eq("")
+    end
+  end
+
+  context "Generating a title for a further education course" do
+    let(:level) { "further_education" }
+
+    it "returns 'Further education'" do
+      expect(generated_title).to eq("Further education")
+    end
+
+    include_examples "with SEND"
+  end
+
+  context "Generating a title for non-further education course" do
+    let(:level) { "primary" }
+
+    context "With a single subject" do
+      context "Example one" do
+        let(:subjects) { [Subject.new(subject_name: "Cats")] }
+        it "Returns the subject name" do
+          expect(generated_title).to eq("Cats")
+        end
+
+        include_examples "with SEND"
+      end
+
+      context "Example two" do
+        let(:subjects) { [Subject.new(subject_name: "Dogs")] }
+        it "Returns the subject name" do
+          expect(generated_title).to eq("Dogs")
+        end
+
+        include_examples "with SEND"
+      end
+    end
+
+    context "With multiple subjects" do
+      context "Example one" do
+        let(:subjects) { [Subject.new(subject_name: "Cats"), Subject.new(subject_name: "Dogs")] }
+
+        it "Returns a name containing both subjects" do
+          expect(generated_title).to eq("Cats with Dogs")
+        end
+
+        include_examples "with SEND"
+      end
+
+      context "Example two" do
+        let(:subjects) { [Subject.new(subject_name: "Dogs"), Subject.new(subject_name: "Cats")] }
+
+        it "Returns a name containing both subjects" do
+          expect(generated_title).to eq("Dogs with Cats")
+        end
+
+        include_examples "with SEND"
+      end
+    end
+
+    context "With modern languages" do
+      context "with one language" do
+        let(:subjects) { [modern_languages, Subject.new(subject_name: "Cats")] }
+
+        it "Returns a name modern language with language" do
+          expect(generated_title).to eq("Modern Languages (Cats)")
+        end
+
+        include_examples "with SEND"
+      end
+
+      context "with two languages" do
+        let(:subjects) do
+          [
+            modern_languages,
+            Subject.new(subject_name: "Cats"),
+            Subject.new(subject_name: "Dogs"),
+          ]
+        end
+
+        it "Returns a name modern language with both languages" do
+          expect(generated_title).to eq("Modern Languages (Cats and Dogs)")
+        end
+
+        include_examples "with SEND"
+      end
+
+      context "with three or more languages" do
+        let(:subjects) do
+          [
+            modern_languages,
+            Subject.new(subject_name: "Cats"),
+            Subject.new(subject_name: "Dogs"),
+            Subject.new(subject_name: "Ducks"),
```

Comment:
This confusing, esp with the service checking if there are 3 "subjects" when generating the title. Shouldn't the languages be in the title?

Reply:
Languages should be in the title when there's 3. eg "Modern languages (French, German, Spanish)"
When there's more than 3 that's too many and we default to "Modern languages"

Reply:
F, G, S

looks alphabetical... is required?

Reply:
Also, the test here has 3, but expects none to be in the title. I think this is related to the confusion of testing the length of `subjects` when we mean `languages` in the service.